homeassistant/components/media_player/denon.py

```diff
@@ -122,17 +152,28 @@ def is_volume_muted(self):
     @property
     def source_list(self):
         """List of available input sources."""
-        return list(self._source_list.keys())
+        # TODO: Add filtering/rename posibility?
+        return sorted(list(self._source_list.keys()))
 
     @property
     def media_title(self):
-        """Current media source."""
-        return self._mediasource
+        """Current media info."""
+        return self._mediainfo
 
     @property
     def supported_media_commands(self):
         """Flag of media commands that are supported."""
-        return SUPPORT_DENON
+        if self._mediasource in MEDIA_MODES.values():
+            return SUPPORT_DENON | SUPPORT_MEDIA_MODES
+        else:
+            return SUPPORT_DENON
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        for pretty_name, name in self._source_list.items():
```

Comment:
Generator expressions and iterators are standard python, so I think it's good to use those when there's a use case, and it will use less lines.